### DISCUSSION TITLE
I is possible to open new windows in the same directory as the focused window (same way new tabs work)?

### DISCUSSION THREAD

**Discussion by TylerBarnes** [wezterm/wezterm#3718] [Q&A] [answered]
The default cwd feels inconsistent to me and I'm hoping there's a configuration setting I can change to fix it.

Opening a new tab opens in the same directory as the tab that was focused when the new tab was opened.
Opening a new window opens in the home directory unless opening from a new tab that was opened in a directory different than the home directory. Then the new window is in the directory the new tab was opened in and not the home directory.

Ideally I'd like new tabs and windows to behave the same way with both opening in the directory the focused window/tab was in when they were opened.

Is there some configuration that will achieve this? I couldn't find anything in the docs

**Comment by wez** [comment, marked as answer]
https://wezfurlong.org/wezterm/config/lua/config/default_cwd.html explains how the working directory is chosen

**Reply by TylerBarnes** [reply]
Awesome, thank you! I looked at this before but missed the part about needing to implement osc7 (I'm on macos)

I googled zsh osc7 and saw this https://github.com/yenaras/zsh/blob/main/osc7_escape_sequence

Then tweaked it to use the code in the docs:

```zsh
_urlencode() {
	local length="${#1}"
	for (( i = 0; i < length; i++ )); do
		local c="${1:$i:1}"
		case $c in
			%) printf '%%%02X' "'$c" ;;
			*) printf "%s" "$c" ;;
		esac
	done
}

osc7_cwd() {
	printf '\033]7;file://%s%s\e\\' "$HOSTNAME" "$(_urlencode "$PWD")"
}

autoload -Uz add-zsh-hook
add-zsh-hook -Uz chpwd osc7_cwd
```

Now it's working perfectly.
Thanks @wez !

**Reply by TylerBarnes** [reply]
If any neovim users read this in the future: you can add something like this to `init.lua` so new tabs/windows open in nvim's cwd:

```lua
function url_encode(str)
  if str then
    str = str:gsub("\n", "\r\n")
    str = str:gsub("([^%w %-%_%.%~])",
      function(c) return string.format("%%%02X", string.byte(c)) end)
    str = str:gsub(" ", "+")
  end
  return str
end

function _wezterm_osc7_cwd()
  local hostname = vim.loop.os_uname().nodename
  local cwd = vim.fn.getcwd()
  local encoded_cwd = url_encode(cwd)

  local osc7_seq = string.format('\027]7;file://%s/%s\027\\', hostname, encoded_cwd)

  -- Use io.write to send the sequence
  io.write(osc7_seq)
end

-- Set up the autocmd to run the function whenever the directory changes
vim.api.nvim_exec([[
    augroup OSC7
        autocmd!
        autocmd DirChanged * lua _wezterm_osc7_cwd()
    augroup END
]], false)
```

**Reply by TylerBarnes** [reply]
One more addition for anyone who comes here looking for this. You can also update the terminal tab title at the same time when nvim changes dirs by modifying the above code to this:

```lua
function _wezterm_osc_cwd()
  local hostname = vim.loop.os_uname().nodename
  local cwd = vim.fn.getcwd()
  local encoded_cwd = url_encode(cwd)

  -- The final sequence to be sent
  local osc7_seq = string.format(
    '\027]7;file://%s/%s\027\\',
    hostname,
    encoded_cwd
  )

  -- Use io.write to send the sequence
  io.write(osc7_seq)

  local cwdDirname = vim.fn.fnamemodify(cwd, ':t')

  local osc1_seq = string.format(
    '\x1b]2;' .. ' ' .. cwdDirname .. '\x1b\\'
  )

  -- Use io.write to send the sequence
  io.write(osc1_seq)
end

-- Set up the autocmd to run the function whenever the directory changes
vim.api.nvim_exec([[
    augroup OSC7
        autocmd!
        autocmd DirChanged * lua _wezterm_osc_cwd()
    augroup END
]], false)
```

**Reply by ursuscamp** [reply]
The above snippet was a great guide, thanks! I also added `SessionLoadPost` to the list of events since loading a session didn't seem to trigger the `DirChanged` event. Now I can jump between sessions and new splits will load in the directory of the new session.

**Reply by mosheavni** [reply]
Honestly this works the same for me:

```lua
vim.api.nvim_create_autocmd('DirChanged', {
  callback = function()
    local cwd = vim.fn.getcwd()
    local hostname = vim.fn.hostname()
    os.execute('printf "\\033]7;file://' .. hostname .. cwd .. '\\033\\\\"')
  end,
})
```

**Reply by sminrana** [reply]
it works perfectly, just add this on your autocmds.lua if you're using LazyVim.

**Comment by wout** [comment]
For anyone using bash, this works as expected:

```bash
# URL encode function
_urlencode() {
    local length="${#1}"
    for (( i = 0; i < length; i++ )); do
        local c="${1:$i:1}"
        case $c in
            [/:%]) printf '%%%02X' "'$c" ;;
            *) printf "%s" "$c" ;;
        esac
    done
}

# Function to send OSC 7 escape sequence
osc7_cwd() {
    printf '\e]7;file://%s%s\e\\' "$HOSTNAME" "$(_urlencode "$PWD")"
}

# Add the osc7_cwd function to PROMPT_COMMAND for bash
PROMPT_COMMAND="osc7_cwd; $PROMPT_COMMAND"
```
